util/src/main/java/org/datacommons/util/CoordinatesResolver.java

```diff
@@ -0,0 +1,112 @@
+package org.datacommons.util;
+
+import java.io.IOException;
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.datacommons.proto.Mcf.McfGraph.PropertyValues;
+import org.datacommons.proto.Mcf.McfGraph.Values;
+import org.datacommons.proto.Mcf.ValueType;
+import org.datacommons.proto.Recon.ResolveCoordinatesRequest;
+import org.datacommons.proto.Recon.ResolveCoordinatesRequest.Coordinate;
+import org.datacommons.proto.Recon.ResolveCoordinatesResponse;
+
+/** Resolves nodes with lat-lngs by calling the DC coordinates resolution API. */
+public class CoordinatesResolver {
+  private final AtomicBoolean resolved = new AtomicBoolean(false);
+
+  private final Set<Coordinate> resolveCoordinates = ConcurrentHashMap.newKeySet();
+
+  private final ConcurrentHashMap<Coordinate, Set<String>> resolvedCoordinates =
+      new ConcurrentHashMap<>();
+
+  private final ReconClient client;
+
+  public CoordinatesResolver(ReconClient client) {
+    this.client = client;
+  }
+
+  public void submitNode(PropertyValues node) {
+    if (resolved.get()) {
+      throw new IllegalStateException("submitNode called after remote resolution.");
+    }
+    getCoordinate(node).ifPresent(resolveCoordinates::add);
+  }
+
+  // TODO: Pick the ID based on a preferred list.
+  public String resolveNode(PropertyValues node) {
```

Comment:
Not fully clear what resolveNode does, maybe document public API usage?

Reply:
I've added comments on the class. The idea is that the resolver will work in 3 stages - submitNode, execute and getResolvedNode. The latter 2 were called remoteResolve and resolveNode in the earlier commit.

Once this PR is submitted, I'll create a `Resolver` base class and have different types of resolvers (entities, coordinates and others like name in the future) as concrete resolvers. The `ExternalIdResolver` will then delegate to these.